### PR TITLE
Generate deb/rpm packages as part of CI

### PR DESCRIPTION
**Pull request by superm1**
This moves the packaging for Debian into contrib/ and runs the CI build to generate the package at the same time.

I would expect that actually in Debian it would mostly be a `git mv contrib/debian.in debian` in the Debian Git repo, and maybe carry some backported patches but otherwise should be the same.  Organizing it this way should make changes needed to packaging easier to identify sooner too.

I think doing the same thing for Fedora makes sense too, but i'm not as experienced with that packaging, so Debian first.

Packages aren't archived after build for distribution in any way, but that's an open discussion item as a possibility right now too.

### CODE REVIEW

**Comment by hughsie** [comment]
I'm not convinced with debian.in -- the .in suffix is normally reserved for files the build system is going to replace a substitution in. The general concept makes sense, and I think it makes sense to do for rpm too -- which https://github.com/hughsie/fwupd/issues/134 would address.

**Comment by superm1** [comment]
`debian.in `made sense when I was storing a `debian/changelog.in `in there in an earlier revision.  I later decided there is no value in that being upstream.  It should be kept on the Debian packaging git repo at [git.debian.org](https://git.debian.org).
I'll adjust it to just `contrib/debian `then and squash this into less commits.

**Comment by superm1** [comment]
OK i've added support for Fedora now too.
Can you take a look at how I did the rpm build and make sure it looks sane?

The other bit is that the build now takes almost 10 minutes on Travis CI.  Considering build/test are essentially ran in packaged form for both distros the two extra builds in unpackaged form may be unnecessary now.

File: contrib/debian/rules [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/make -f
+# -*- makefile -*-
+
+# pie causes build failures
+export DEB_BUILD_MAINT_OPTIONS = hardening=+all,-pie
```
Comment (hughsie) [inline contrib/debian/rules]: PIE causes build failures? Got a link? I think PIE is a really good idea.

File: contrib/debian/libfwupd1.symbols [hunk 2]
```diff
@@ -0,0 +1,104 @@
+libfwupd.so.1 libfwupd1 #MINVER#
+ fwupd_client_clear_results@Base 0.7.0
```
Comment (hughsie) [inline contrib/debian/libfwupd1.symbols]: Could this go into libfwupd/fwupd.map instead?

**Comment by hughsie** [comment]
> the two extra builds in unpackaged form may be unnecessary

Are the tests being run in both rpm and deb-installing scripts? I think an explicit "ninja test" is needed for the rpm build at least.

File: contrib/debian/rules (hunk 1 above)
Comment (superm1) [inline contrib/debian/rules]: Yeah I'd like to keep it  enabled, but when switching to meson/ninja this came up and that was the temporary solution.
I'll turn it off on the next upload to experimental to confirm if it's still a problem (test build on x86-64 seems to indicate that it shouldn't be a problem right now).

**Comment by superm1** [comment]
> Are the tests being run in both rpm and deb-installing scripts? I think an explicit "ninja test" is needed for the rpm build at least.

For deb yes, debhelper handles it.  For rpm, I'll add something to let it turn on conditionally.

File: contrib/debian/libfwupd1.symbols (hunk 2 above)
Comment (superm1) [inline contrib/debian/libfwupd1.symbols]: The symbols system (https://www.debian.org/doc/debian-policy/ch-sharedlibs.html#s-sharedlibs-symbols) is meant to live and work specifically at the packaging level.

Comment (hughsie) [inline contrib/debian/libfwupd1.symbols]: okay, fair enough!

**Comment by hughsie** [comment]
LGTM.

**Comment by superm1** [comment]
OK thanks.  I'm just making sure I fixed the failed build introduced by  e691d30  (I expect 559691c should fix it)

**Pull request merged**